sleap_io/io/nwb.py

```diff
@@ -6,27 +6,298 @@
+    if not isinstance(labels, Labels):
+        raise ValueError("The input must be a SLEAP Labels object.")
+    
+    skeletons_list: list[Skeleton] = [] # type: ignore[assignment]
+    skeletons = Skeletons(skeletons=skeletons_list) # type: ignore[assignment]
+    training_frame_list = []
+    for i, labeled_frame in enumerate(labels.labeled_frames):
+        training_frame_name = name_generator("training_frame")
+        training_frame_annotator = f"{training_frame_name}_{i}"
+        skeleton_instances_list = []
+        
+        for instance in labeled_frame.instances:
+            if isinstance(instance, PredictedInstance):
+                continue
+            
+            if instance.skeleton not in skeletons_list:
+                skeletons_list.append(instance.skeleton)
+                skeletons = Skeletons(skeletons=skeletons_list)
+            skeleton_instance = instance_to_skeleton_instance(instance)
+            skeleton_instances_list.append(skeleton_instance)
+
+        training_frame_skeleton_instances = SkeletonInstances(
+            skeleton_instances=skeleton_instances_list
+        )
+        training_frame_video = labeled_frame.video
+        training_frame_video_index = labeled_frame.frame_idx
+
+        if img_paths:
+            source_video = ImageSeries(
+                name=training_frame_name,
+                description=training_frame_annotator,
+                unit="NA",
+                format="external",
+                external_file=img_paths,
+                dimension=[
+                    training_frame_video.backend.img_shape[0],
+                    training_frame_video.backend.img_shape[1],
+                ],
+                starting_frame=[0],
+                rate=30.0, # change to `video.backend.fps` when available
+            )
+        else:
+            source_video=ImageSeries(
+                name=training_frame_name,
+                description=training_frame_annotator,
+                unit="NA",
+                format="external",
+                external_file=[training_frame_video.filename],
+                dimension=[
+                    training_frame_video.shape[1],
+                    training_frame_video.shape[2],
+                ],
+                starting_frame=[0],
+                rate=30.0, # change to `video.backend.fps` when available
+            )
+        training_frame = TrainingFrame(
+            name=training_frame_name,
+            annotator=training_frame_annotator,
+            skeleton_instances=training_frame_skeleton_instances,
+            source_video=source_video,
+            source_video_frame_index=training_frame_video_index,
+        )
+        training_frame_list.append(training_frame)
+
+    training_frames = TrainingFrames(training_frames=training_frame_list)
+    pose_training = PoseTraining(
+        training_frames=training_frames,
+        source_videos=videos_to_source_videos(labels.videos),
+    )
+    return pose_training
```

Comment:
The whole way videos are being handled here needs to be refactored.

Start with just the second case (pointing to an external video file), but as with the skeletons, create a mapping between SLEAP `Video` objects and NWB `ImageSeries` so that as you're looping through the labeled frames, you can reference the exact object.

For the images, we'll need a very different routine since we have to save them, have a mapping of `(video, frame_idx)` --> `img_path`, create a single `ImageSeries`, and reference the appropriate index within that list.